### PR TITLE
Fixes #22 [ Custom refresh rates for different sensors ]

### LINKED ISSUE #22: Give users the freedom to set the sensor update interval per sensor basis
Currently the sensors have update interval hardcoded. This is required to be abstracted out such that the users can set them according to their own requirement.

### PR DESCRIPTION
**Pull request by vinusankars**
Refresh rates are input by the user using argparse.
The GUI gets the refresh rates for sensor and uses it to update the sensor rates.

### CODE REVIEW

**Comment by darxtrix** [comment]
Looks good, we might need to add a `.ptoprc` file saving the default configuration for ptop invocation so that the user don't require to pass them each time.

**Comment by vinusankars** [comment]
@darxtrix  can you please make me a main contributor? I will try incorporating a config file for keyboard shortcuts too!

**Comment by darxtrix** [comment]
Sure, but I would like to take a little time before that. We have to define and streamline the PR body/heading format and the commit messages format also.

File: ptop/main.py [hunk 1]
```diff
@@ -28,23 +28,127 @@ def main():
...
         # app wide global stop flag
         global_stop_event = threading.Event()
 
-        s = Statistics(SENSORS_LIST,global_stop_event)
+        s = Statistics(SENSORS_LIST,global_stop_event,sensor_rate)
         # internally uses a thread Job 
         s.generate()
         logger.info('Statistics generating started')
-        app = PtopGUI(s.statistics,global_stop_event,theme)
+        app = PtopGUI(s.statistics,global_stop_event,theme,sensor_rate)
```
Comment (darxtrix) [inline ptop/main.py]: Why do we require it in here, only passing to `Statistics` class will do the job.

File: ptop/main.py [hunk 2]
```diff
@@ -28,23 +28,127 @@ def main():
                                  simple
                                  blackonwhite
                             ''')
+
+        parser.add_argument('-cpu',
+                            dest='cpu',
```
Comment (darxtrix) [inline ptop/main.py]: please use some expressive names like `cpuSensorResponseTime` or short forms like `csrt`. Its better for understanding and maintaining.

File: ptop/main.py [hunk 3]
```diff
@@ -28,23 +28,127 @@ def main():
...
+                            required=False,
+                            help=
+                            '''
+                                Input sensor
+        						update interval in
+        						milli seconds less than 1000.
+        						For example 500
+                            ''')
+
+        parser.add_argument('-disk',
```
Comment (darxtrix) [inline ptop/main.py]: same here, naming fix.

File: ptop/main.py (hunk 1 above)
Comment (vinusankars) [reply ptop/main.py]: The sensor gets the data in specified interval, but it's not displayed in GUI accordingly. To do the same the update rate is sent to PtopGUI.

File: ptop/main.py [hunk 4]
```diff
@@ -28,23 +28,127 @@ def main():
...
                             action='version',
                             version='ptop {}'.format(__version__))
 
         results = parser.parse_args()
+
         if results.theme:
             theme = results.theme
         else:
             theme = 'elegant'
+
```
Comment (darxtrix) [inline ptop/main.py]: Try using ternary expressions to reduce the size of the code and also helper functions.

File: ptop/statistics/statistics.py [hunk 5]
```diff
@@ -29,9 +30,9 @@ def generate(self):
         '''
             Generate the stats using the plugins list periodically
         '''
-        for sensor in self.plugins:
+        for i, sensor in enumerate(self.plugins):
```
Comment (darxtrix) [inline ptop/statistics/statistics.py]: You are passing in alphabetical order for now. We can do better if you pass the rate in a dictionary and use the key as the sensor name that you can specify in the sensor class definition. That's less error prune and scalable way of doing things. Let  me know if you don't get me.

File: ptop/main.py (hunk 1 above)
Comment (darxtrix) [reply ptop/main.py]: We can assume that the GUI refresh rate is minimum compared to other sensor response times but its fine as I see you are taking minimum of all. Should we make a gitter channel of all interested folks interested in contributing ?

Comment (vinusankars) [reply ptop/main.py]: Sure.

File: ptop/main.py (hunk 2 above)
Comment (vinusankars) [reply ptop/main.py]: Please if it works now.

File: ptop/statistics/statistics.py (hunk 5 above)
Comment (vinusankars) [reply ptop/statistics/statistics.py]: See now please.

**Comment by vinusankars** [comment]
@darxtrix  We can try to deb package ptop 1.0 too, in the coming days. (apt-get)

**Comment by darxtrix** [comment]
@vinusankars Absolutely, there is an issue tracking this feature https://github.com/darxtrix/ptop/issues/23. Additionally we can have a brew tap also. I remember someone created an aur package for arch based systems, it have to be updated now. Let's finalize a date for the release ?

**Comment by vinusankars** [comment]
@darxtrix What about tomorrow?

**Comment by darxtrix** [comment]
No that will be too soon. Need to test ptop well before that. I am using the *updated* ptop on daily basis for the task managing task now. Lets setup a session of 30 mins on sometime Friday and test ptop in different scenarios and take a buffer day to fix issues if there are some. I will write changelog and a blog on Sunday. Then we can go live on next Monday ?

**Comment by vinusankars** [comment]
Sure. @deeps-nars  please see the above comments. Let us do this.

**Comment by darxtrix** [comment]
@vinusankars @deeps-nars You guys please join gitter. Its better to keep track of meaningful conversations over there otherwise we will have to lurk through all the issues/PRs for finding comments :sweat_smile:

**Comment by darxtrix** [comment]
Merged with some code refactoring, great work @vinusankars !

**Pull request merged**